apps/server/src/modules/common-cartridge/common-cartridge-client/lesson-client/mapper/lesson-dto.mapper.ts

```diff
@@ -71,58 +56,60 @@ export class LessonDtoMapper {
 		switch (lessonContentResponse.component) {
-			case LessonContentResponseComponent.TEXT:
+			case LessonContentResponseComponentEnum.Text:
 				return new LessonContentDto({
 					id: lessonContentResponse.id,
 					title: lessonContentResponse.title,
 					component: lessonContentResponse.component,
 					hidden: lessonContentResponse.hidden,
 					content: new ComponentTextPropsDto(lessonContentResponse.content as ComponentTextPropsImpl),
 				});
-			case LessonContentResponseComponent.ETHERPAD:
+			case LessonContentResponseComponentEnum.Etherpad:
 				return new LessonContentDto({
 					id: lessonContentResponse.id,
 					title: lessonContentResponse.title,
 					component: lessonContentResponse.component,
 					hidden: lessonContentResponse.hidden,
 					content: new ComponentEtherpadPropsDto(lessonContentResponse.content as ComponentEtherpadPropsImpl),
 				});
-			case LessonContentResponseComponent.GEO_GEBRA:
+			case LessonContentResponseComponentEnum.GeoGebra:
 				return new LessonContentDto({
 					id: lessonContentResponse.id,
 					title: lessonContentResponse.title,
 					component: lessonContentResponse.component,
 					hidden: lessonContentResponse.hidden,
 					content: new ComponentGeogebraPropsDto(lessonContentResponse.content as ComponentGeogebraPropsImpl),
 				});
-			case LessonContentResponseComponent.INTERNAL:
+			case LessonContentResponseComponentEnum.Internal:
 				return new LessonContentDto({
 					id: lessonContentResponse.id,
 					title: lessonContentResponse.title,
 					component: lessonContentResponse.component,
 					hidden: lessonContentResponse.hidden,
 					content: new ComponentInternalPropsDto(lessonContentResponse.content as ComponentInternalPropsImpl),
 				});
-			case LessonContentResponseComponent.RESOURCES:
+			case LessonContentResponseComponentEnum.Resources:
 				return new LessonContentDto({
 					id: lessonContentResponse.id,
 					title: lessonContentResponse.title,
 					component: lessonContentResponse.component,
 					hidden: lessonContentResponse.hidden,
-					content: new ComponentLernstorePropsDto(lessonContentResponse.content as ComponentLernstorePropsImpl),
-				});
-			case LessonContentResponseComponent.NEX_BOARD:
-				return new LessonContentDto({
-					id: lessonContentResponse.id,
-					title: lessonContentResponse.title,
-					component: lessonContentResponse.component,
-					hidden: lessonContentResponse.hidden,
-					content: new ComponentNexboardPropsDto(lessonContentResponse.content as ComponentNexboardPropsImpl),
+					content: this.mapContentOfLernstoreElement(lessonContentResponse),
 				});
 			default:
 				return null;
 		}
 	}
+
+	private static mapContentOfLernstoreElement(
+		lessonContentResponse: LessonContentResponse
+	): ComponentLernstorePropsDto {
+		if (!lessonContentResponse.content) {
+			return { resources: [] };
+		}
+
+		return new ComponentLernstorePropsDto(lessonContentResponse.content as ComponentLernstorePropsImpl);
+	}
```

Comment:
Could it work like that?

```suggestion
	private static mapContentOfLernstoreElement(
        lessonContentResponse: LessonContentResponse
    ): ComponentLernstorePropsDto {
        if (!lessonContentResponse.content) {
            return new ComponentLernstorePropsDto([]);
        }

        const lernstoreContent = lessonContentResponse.content as ComponentLernstorePropsImpl;
        const resources = lernstoreContent.resources.map((resource) => new LernstoreResourcesDto(resource));
        return new ComponentLernstorePropsDto(resources);
    }
```